src/map.cpp

```diff
@@ -1944,7 +1951,16 @@ bool ServerMap::saveBlock(MapBlock *block, MapDatabase *db)
 	o.write((char*) &version, 1);
 	block->serialize(o, version, true);
 
-	bool ret = db->saveBlock(p3d, o.str());
+	bool ret;
+	try {
+		ret = db->saveBlock(p3d, o.str());
+	} catch (DatabaseException &e) {
+		could_not_save = true;
```

Comment:
This isn't necessarily only when the DB is readonly, I'm not sure `could_not_save` makes sense. For example, is there a DB exception on timeouts? Could certain backends cause timeouts when backups are running?